### PR TITLE
Update Dfsu documentation

### PR DESCRIPTION
**Pull request by jsmariegaard**

### CODE REVIEW

**Comment by jsmariegaard** [comment]
@ecomodeller , how should we now (with .md instead of .rst) make "warning" boxes? (see my attempt in the checked-in code)

**Comment by ecomodeller** [comment]
image

renders

image

File: docs/data-structures.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Data Structures
+
+MIKE IO has these primary data structures: 
+
+* [Dataset](Dataset) - a collection of DataArrays corresponding to the contents of a dfs file; typically obtained from `mikeio.read()`
+* [DataArray](DataArray) - data and meta data corresponding to one "item" in a dfs file. 
+* **Geometry** - spatial description of the data in a dfs file; comes in different flavours: [Grid1D](Grid1D), [Grid2D](Grid2D), [Grid3D](Grid3D), [GeometryFM](GeometryFM), [GeometryFM3D](GeometryFM3D), etc. corresponding to different types of dfs files. 
+* **Dfs** - an object returned by `dfs = mikeio.open()` containing the meta data (=header) of a dfs file ready for reading the data (which can be done with `dfs.read()`); exists in different specialized versions: [Dfs0](Dfs0), [Dfs1](Dfs1), [Dfs2](Dfs2), [Dfs3](Dfs3), [Dfsu2DH](Dfsu2DH), [Dfsu3D](Dfsu3D), [Dfsu2DV](Dfsu2DV), [DfsuSpectral](DfsuSpectral),
```
Comment (ecomodeller) [inline docs/data-structures.md:8]:
```suggestion
* **Dfs** - an object returned by `dfs = mikeio.open()` containing the metadata (=header) of a dfs file ready for reading the data (which can be done with `dfs.read()`); exists in different specialized versions: [Dfs0](Dfs0), [Dfs1](Dfs1), [Dfs2](Dfs2), [Dfs3](Dfs3), [Dfsu2DH](Dfsu2DH), [Dfsu3D](Dfsu3D), [Dfsu2DV](Dfsu2DV), [DfsuSpectral](DfsuSpectral), 
```

File: docs/dataarray.md [hunk 2]
```diff
@@ -0,0 +1,154 @@
+# DataArray
+
+The [DataArray](DataArray) is the common MIKE IO data structure 
+for *item* data from dfs files. 
+The `mikeio.read()` methods returns a Dataset as a container of DataArrays (Dfs items)
+
+Each DataArray have the following properties:
+* **item** - an  [`ItemInfo`](ItemInfo) with name, type and unit
+* **time** - a pandas.DateTimeIndex with the time instances of the data
```
Comment (ecomodeller) [inline docs/dataarray.md:9]:
```suggestion
* **time** - a [pandas.DateimeIndex](https://pandas.pydata.org/docs/reference/api/pandas.DatetimeIndex.html) with the time instances of the data
```

File: docs/dfs-overview.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+* start_time - First time instance (as datetime)
+* end_time - Last time instance (as datetime)
+* geometry - spatial description of the data in the file ([Grid1D](Grid1D), [Grid2D](Grid2D), etc ... )
+* deletevalue - File delete value (NaN value)
+
+
+
+## Open or read? 
+
+Dfs files contain data and meta data.
```
Comment (ecomodeller) [inline docs/dfs-overview.md:23]:
```suggestion
Dfs files contain data and metadata. 
```

File: docs/dfsu-2dv-vertical-profile.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
+# Dfsu 2DV Vertical Profile
+
+
+Apart from the common [dfsu-geometry properties and methods](./dfu-mesh-overview.md#mike-io-flexible-mesh-geometry), Dfsu2DV has the below additional *properties* (from it's geometry [GeometryFMVerticalProfile](GeometryFMVerticalProfile)):
```
Comment (ecomodeller) [inline docs/dfsu-2dv-vertical-profile.md:4]:
```suggestion
In addition to the common [dfsu-geometry properties and methods](./dfu-mesh-overview.md#mike-io-flexible-mesh-geometry), Dfsu2DV has the below additional *properties* (from it's geometry [GeometryFMVerticalProfile](GeometryFMVerticalProfile)): 
```

File: docs/dfsu-2dv-vertical-profile.md [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    mikeio.dfsu_layered.Dfsu2DV.layer_ids
+    mikeio.dfsu_layered.Dfsu2DV.top_elements
+    mikeio.dfsu_layered.Dfsu2DV.bottom_elements
+    mikeio.dfsu_layered.Dfsu2DV.n_layers_per_column
+    mikeio.dfsu_layered.Dfsu2DV.e2_e3_table
+    mikeio.dfsu_layered.Dfsu2DV.elem2d_ids
+```
+
+
+And apart from the basic dfsu functionality, Dfsu2DV has the below additional *methods*:
```
Comment (ecomodeller) [inline docs/dfsu-2dv-vertical-profile.md:24]:
```suggestion
And in addition to the basic dfsu functionality, Dfsu2DV has the below additional *methods*: 
```

File: docs/dfsu-3d.md [hunk 6]
```diff
@@ -0,0 +1,60 @@
+# Dfsu 3D
+
+
+Apart from the common [dfsu-geometry properties and methods](./dfu-mesh-overview.md#mike-io-flexible-mesh-geometry), Dfsu3D has the below additional *properties* (from it's geometry [GeometryFM3D](GeometryFM3D)):
```
Comment (ecomodeller) [inline docs/dfsu-3d.md:4]:
```suggestion
In addition to the common [dfsu-geometry properties and methods](./dfu-mesh-overview.md#mike-io-flexible-mesh-geometry), Dfsu3D has the below additional *properties* (from it's geometry [GeometryFM3D](GeometryFM3D)): 
```

File: docs/dfsu-3d.md [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+    mikeio.dfsu_layered.Dfsu3D.top_elements
+    mikeio.dfsu_layered.Dfsu3D.bottom_elements
+    mikeio.dfsu_layered.Dfsu3D.n_layers_per_column
+    mikeio.dfsu_layered.Dfsu3D.geometry2d
+    mikeio.dfsu_layered.Dfsu3D.e2_e3_table
+    mikeio.dfsu_layered.Dfsu3D.elem2d_ids
+```
+
+
+And apart from the basic dfsu functionality, Dfsu3D has the below additional *methods*:
```
Comment (ecomodeller) [inline docs/dfsu-3d.md:23]:
```suggestion
And in addition to from the basic dfsu functionality, Dfsu3D has the below additional *methods*: 
```

File: docs/mesh.md [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+
+```python
+>>> msh = mikeio.open("../tests/testdata/odense_rough.mesh")
+>>> msh
+Number of elements: 654
+Number of nodes: 399
+Projection: UTM-33
+```
+
+Apart from the common [dfsu-geometry properties and methods](./dfu-mesh-overview.md#mike-io-flexible-mesh-geometry), `Mesh` has the following properties and methods:
```
Comment (ecomodeller) [inline docs/mesh.md:16]:
```suggestion
In addition to the common [dfsu-geometry properties and methods](./dfu-mesh-overview.md#mike-io-flexible-mesh-geometry), `Mesh` has the following properties and methods:
```

File: mikeio/spatial/grid_geometry.py [hunk 9]
```diff
@@ -478,7 +480,8 @@ def contains(self, coords):
 
     def find_index(self, x: float = None, y: float = None, coords=None, area=None):
         """Find nearest index (i,j) of point(s)
-           -1 is returned if point is outside grid
+
+        -1 is returned if point is outside grid
```
Comment (ecomodeller) [inline mikeio/spatial/grid_geometry.py:484]: Not exactly part of this PR, but this is not consistent with [list.index](https://docs.python.org/3/tutorial/datastructures.html) which raises ValueError if the value you are looking for is not in the list.

**Review by ecomodeller — APPROVED** [review]
📖 🤓 :shipit:

File: mikeio/spatial/grid_geometry.py (hunk 9 above)
Comment (jsmariegaard) [reply mikeio/spatial/grid_geometry.py:484]: Right, I guess the current behavior is better if you have many points (or an area) that you finding indices for...

**Pull request merged**